### ISSUE TITLE
Docker client API version is too old

### ISSUE THREAD

**Issue by jkingyens** [ibuildthecloud/systemd-docker#33]
I'm getting the following error with the docker version shipped with CoreOS alpha images (1.7.1): 

"client is too old, minimum supported API version is 1.12"

Not sure if the fix is trivial (just a version bump) or if more care is needed to update the API calls.

**Comment by trnubo** [comment]
Just hit the same issue with CentOS 7.1

```
systemd-docker[14769]: 2015/08/14 03:23:44 API error (400): client is too old, minimum supported API version is 1.12, please upgrade your client to a newer version
```
```
[centos@centos ~]$ cat /etc/centos-release 
CentOS Linux release 7.1.1503 (Core) 
[centos@centos ~]$ rpm -q docker
docker-1.7.1-108.el7.centos.x86_64
```

**Comment by hinnerk** [comment]
The problem escalated to CoreOS Beta Channel in the last week. Is this software still actively maintained @ibuildthecloud ?

**Comment by renecunningham** [comment]
Seeing this with CentOS 7.1 too.

```
# lsb_release -a
LSB Version:    :core-4.1-amd64:core-4.1-noarch:cxx-4.1-amd64:cxx-4.1-noarch:desktop-4.1-amd64:desktop-4.1-noarch:languages-4.1-amd64:languages-4.1-noarch:printing-4.1-amd64:printing-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.1.1503 (Core) 
Release:        7.1.1503
Codename:       Core

# rpm -q docker
docker-1.7.1-108.el7.centos.x86_64

# cd /opt/systemd-docker && git log --pretty=format:'%h' -n 1
58b2620

# /opt/systemd-docker/bin/systemd-docker --cgroups name=systemd --env run --rm --publish=9100:9000 --volume=redacted:redacted:z --name redacted redacted
2015/08/28 01:08:37 API error (400): client is too old, minimum supported API version is 1.12, please upgrade your client to a newer version
```

Work around for me right now is to downgrade to 1.6.2.

```
# yum downgrade docker-1.6.2-14.el7.centos
# systemctl restart docker
```

**Comment by agend07** [comment]
all u have to do is to change 1.11 to 1.12 in line 265 main.go - which few people already did in their own forks (ex: github.com/agend07/systemd-docker)

then use build script and use the binary

**Comment by Allezxandre** [comment]
@agend07 Thanks! It worked for me

**Comment by hookenz** [comment]
What happens if you have a generic unit file and people grab that and run it on docker 1.6 after you've updated to 1.12 on main.go?  Will it break and crash into a horrible mess?

**Comment by larsdecker** [comment]
The current API Version is greater than 1.19. It seems that the server will also support some old version. The updated api-version in the main.go is the minimum api version.

docker version
Client version: 1.7.1
Client API version: 1.19
Go version (client): go1.4.2
Git commit (client): 2c2c52b-dirty
OS/Arch (client): linux/amd64
Server version: 1.7.1
Server API version: 1.19
Go version (server): go1.4.2
Git commit (server): 2c2c52b-dirty
OS/Arch (server): linux/amd64

**Comment by hookenz** [comment]
Thanks.  I've decided not to use systemd-docker for fear of breaking things later.  It's a hack that isn't really necessary if you don't mix the use of systemd and docker together.

**Comment by djmaze** [comment]
@hookenz Yes, the new release breaks on Docker 1.6.2 for me. The API version change seems not backwards compatible. 

Keeping the older systemd-docker on servers with the older Docker version for now. Kind of a mess.
